Automattic/simplenote-macos
# Magic Links: The one PR to rule them all

### Fix
... one PR to find them, one PR to bring them all, and in the darkness bind them...

This PR brings together all of the previous changes to magic links authorization that has been made on SNMac and then merges them all back into trunk.  There aren't any unreviewed changes here (although we might make some button copy changes before merging) but we should smoke test the auth views before we merge it back to trunk and then release the new auth to the masses.

fixes: #1187 

### Test
***(Required)*** List the steps to test the behavior.  For example:

Nothing specific here, but we want to test the new auth views so, try and consider this list of things:
1. Load through the different views in the auth process and make sure that going back and forth between views to ensure it looks right and the animations are solid
2. Attempt a sign up
3. Attempt login with code.  While testing code login, try entering bad codes and expired codes
4. Attempt login with email link.  Also test an expired link
5. Attempt login with password
6. Make sure that if you put invalid values in the different text fields that you are given an error

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
Doesn't require updates because it is already there
